packages/next/src/server/typescript/rules/config.ts

```diff
@@ -364,6 +374,70 @@ const config = {
         if (!options) return
         content = options[entryName]
       }
+
+      if (entryName === 'metadata' || entryName === 'generateMetadata') {
+        const sourceFile = getSource(fileName)
+        let start = 0
+        let hasMetadataImport = false
+
+        if (sourceFile) {
+          ts.forEachChild(sourceFile, (node) => {
```

Comment:
Does TS not have a dedicated visitor pattern? Seems wasteful to visit each child.